cmd/olm/main.go

```diff
@@ -131,84 +132,91 @@ func main() {
 	<-ready
 
 	if *writeStatusName != "" {
```

Comment:
Nice, I like this. Would it be useful to us (and other teams) to factor this out into its own module/helper?

Reply:
Maybe? This particular PR is really only about 5 lines long, but I had to shuffle other code around which makes it look bigger. Do you mean all the changed lines or just the discovery part? My understanding is that in the future we'll be outfitting all of OLM to update these status conditions, not just during start up. So it might be best to wait until later to see how that all shakes out.